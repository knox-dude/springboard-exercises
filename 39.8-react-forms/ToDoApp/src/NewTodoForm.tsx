import { useState } from 'react';
import { v4 as uuid } from 'uuid';

interface NewTodoFormProps {
  addTodo: (text: string, id: string) => void;
}

function NewTodoForm({ addTodo }: NewTodoFormProps) {
  const [text, setText] = useState("");
  const id = uuid();

  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    addTodo(text, id);
    setText("");
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        type="text"
        value={text}
        onChange={(e) => setText(e.target.value)}
      />
      <input
        type="text"
        value={text}
        onChange={(e) => setText(e.target.value)}
      />
      <button type="submit">Add Todo</button>
    </form>
  );
}

export default NewTodoForm;