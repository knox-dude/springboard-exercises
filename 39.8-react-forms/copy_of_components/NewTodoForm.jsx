import { useState } from "react";
import propTypes from "prop-types"
import { v4 as uuid } from "uuid";

function NewTodoForm( { createTodo } ) {
    const [task, setTask] = useState("");

    const handleSubmit = (e) => {
        e.preventDefault();
        createTodo({ task, id: uuid() });
        setTask("");
    };

    return (
        <form onSubmit={handleSubmit}>
            <input
                type="text"
                value={task}
                onChange={(e) => setTask(e.target.value)}
            />
            <button type="submit">Add Todo</button>
        </form>
    );
}

NewTodoForm.propTypes = {
    createTodo: propTypes.func.isRequired,
}

export default NewTodoForm;