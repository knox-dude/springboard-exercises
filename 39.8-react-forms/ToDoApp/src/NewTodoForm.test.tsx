import { render, screen, fireEvent } from "@testing-library/react";
import NewTodoForm from "./NewTodoForm";

describe("NewTodoForm", () => {
  // mock add todo fn
  const addTodo = jest.fn();

  //smoke test
  it("renders without crashing", () => {
    render(<NewTodoForm addTodo={addTodo} />);
  });

  //snapshot test
  it("matches snapshot", () => {
    const { asFragment } = render(<NewTodoForm addTodo={addTodo} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("calls addTodo function", () => {
    render(<NewTodoForm addTodo={addTodo} />);
    const addTodoButton = screen.getByText("Add Todo");
    fireEvent.click(addTodoButton);
    expect(addTodo).toHaveBeenCalled();
  });
});
