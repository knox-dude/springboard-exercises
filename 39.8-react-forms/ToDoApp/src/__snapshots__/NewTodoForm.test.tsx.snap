// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTodoForm matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <input
      placeholder="Enter todo..."
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Add Todo
    </button>
  </form>
</DocumentFragment>
`;
